### ISSUE TITLE
Bug: string is not a string

### ISSUE THREAD

**Issue by HansBrende** [microsoft/TypeScript#60764]
### 🔎 Search Terms

string extends template bug wrong error conditional

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.7.2#code/FAFwngDgpgBAcgexgXhgbQGYEMA2BnKAXWFElgEk8BBAZRACcBLAOwHMAeGgPhRhpigAPEFGYATPDAAGAEgDeLDFHowA+gF8pMAPwwGAV1gAuGNnxQA3CXDQ+DFq0q17bdgBUBw0RJh4XrHlQ0JzomV1k5N00uQk8RcUlEHVNcAhgTAygSAGMEZj9fAAsEfRwxKhwAdywwPAAhKDd6QxNQhxD-dgByPC7AvWbLIA

### 💻 Code

```ts
type No = [false]

type IsAString<S> = S extends `${infer _}` ? true : false;

type StringIsAString<T extends string> = [IsAString<`${T}`>] extends No ? false : true

const shouldAlwaysBeTrue: StringIsAString<'s'> = true;
```

### 🙁 Actual behavior

`shouldAlwaysBeTrue` reports the following error:

> Type 'true' is not assignable to type 'false'.(2322)

### 🙂 Expected behavior

No error

### Additional information about the issue

This is the most minimal reproduction of the issue I could come up with after whittling it down from a _much_ more complicated scenario.

Interestingly enough, if I change _almost anything_ about it, the issue goes away. For example, if I replace `No` with `[false]`, issue goes away. If I take out one of the templates, issue goes away. Etc.

**Comment by MartinJohns** [comment]
Possibly a duplicate of #57650.

**Comment by HansBrende** [comment]
I read through that other issue and I don't see how it could be a duplicate, as I'm not sure how a so-called "permissive instantiation" of `No` could be anything other than `[false]`. But it's also possible that it is the same issue and I'm too stupid to understand the connection 😆

**Comment by KisaragiEffective** [comment]
Workaround: you may unquote `T`:

```ts
type IsAString<S> = S extends `${infer _}` ? true : false;
type StringIsAString<T extends string> = {result: IsAString<T>} extends {result: false} ? false : true
const shouldAlwaysBeTrue: StringIsAString<'s'> = true;
```

**Comment by RyanCavanaugh** [comment]
This is the same premature generic conditional-type evaluation tracked in [#57650](https://github.com/microsoft/TypeScript/issues/57650). Wrapping the type parameter in a template-literal type lets the outer conditional resolve before the parameter is instantiated, so `StringIsAString<'s'>` becomes `false`.

Tracking this behavior at #57650.
